huaweicloud/services/live/resource_huaweicloud_live_recording.go

```diff
@@ -0,0 +1,387 @@
+			},
+
+			"file_naming": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"max_stream_pause_length": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(0, 300),
+				Computed:     true,
+			},
+		},
+	}
+}
+
+func resourceRecordingCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	createBody, err := buildRecordingParams(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	createOpts := &model.CreateRecordRuleRequest{
+		Body: createBody,
+	}
+
+	log.Printf("[DEBUG] Create Live recording params : %#v", createOpts)
+
+	resp, err := client.CreateRecordRule(createOpts)
+	if err != nil {
+		return diag.Errorf("error creating Live recording: %s", err)
+	}
+
+	if resp.Id == nil {
+		return diag.Errorf("error creating Live recording: ID not found in the API response")
+	}
+
+	d.SetId(*resp.Id)
+
+	return resourceRecordingRead(ctx, d, meta)
+}
+
+func resourceRecordingRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	response, err := client.ShowRecordRule(&model.ShowRecordRuleRequest{Id: d.Id()})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Live recording")
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("domain_name", response.PublishDomain),
+		d.Set("app_name", response.App),
+		d.Set("stream_name", response.Stream),
+		d.Set("type", utils.MarshalValue(response.RecordType)),
+		d.Set("obs", flattenObs(response.DefaultRecordConfig)),
+		d.Set("hls", flattenHls(response.DefaultRecordConfig)),
+		d.Set("flv", flattenFlv(response.DefaultRecordConfig)),
+		d.Set("mp4", flattenMp4(response.DefaultRecordConfig)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceRecordingUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	updateBody, err := buildRecordingParams(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updateOpts := &model.UpdateRecordRuleRequest{
+		Id:   d.Id(),
+		Body: updateBody,
+	}
+
+	_, err = client.UpdateRecordRule(updateOpts)
+	if err != nil {
+		return diag.Errorf("error updating Live recording: %s", err)
+	}
+
+	return resourceRecordingRead(ctx, d, meta)
+}
+
+func resourceRecordingDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
replace ctx with '_' as it's not used in this func

Reply:
done